src/Build/Evaluation/IItem.cs

```diff
@@ -52,5 +52,22 @@ string ProjectDirectory
         /// Metadata value is the escaped value initially set.
         /// </summary>
         string GetMetadataValueEscaped(string name);
+
+        /// <summary>
+        /// Returns true if a particular piece of metadata is defined on this item (even if
+        /// its value is empty string) otherwise false.
+        /// This includes built-in metadata and metadata from item definitions.
+        /// </summary>
+        /// <remarks>
+        /// It has to include all of these because it's used for batching, which doesn't
+        /// care where the metadata originated.
+        /// </remarks>
+        bool HasMetadata(string name);
+
+        /// <summary>
+        /// Get the metadata value with the specified key. 
+        /// Return value indicates whether get the metadata or not. Returns false if metadata does not exist.
+        /// </summary>
+        bool TryGetMetadataValueEscaped(string name, out string value);
```

Comment:
Agree with you. Have updated function.

Reply:
I first proposed the TryGet functionality just because I saw "Has_" and "Get_" functions. Good point that the function just returns null on failure, though 👍